File: Muzix_Auth_Service-main/FrontEnd_UI/my-app/src/services/movieService.js
const TMDB_API_KEY = '4833b1d3a2c00e56714bd2905095d5c8'; // Replace with your actual TMDB API key
const TMDB_BASE_URL = 'https://api.themoviedb.org/3';
const IMAGE_BASE_URL = 'https://image.tmdb.org/t/p/w500';

// Get popular movies
export const getPopularMovies = async () => {
  try {
    const response = await fetch(
      `${TMDB_BASE_URL}/movie/popular?api_key=${TMDB_API_KEY}&language=en-US&page=1`
    );
    
    if (!response.ok) {
      throw new Error('Failed to fetch popular movies');
    }
    
    const data = await response.json();
    return data.results;
  } catch (error) {
    console.error('Error fetching popular movies:', error);
    throw error;
  }
};

// Search movies
export const searchMovies = async (query) => {
  try {
    if (!query.trim()) {
      return [];
    }
    
    const response = await fetch(
      `${TMDB_BASE_URL}/search/movie?api_key=${TMDB_API_KEY}&language=en-US&query=${encodeURIComponent(query)}&page=1`
    );
    
    if (!response.ok) {
      throw new Error('Failed to search movies');
    }
    
    const data = await response.json();
    return data.results;
  } catch (error) {
    console.error('Error searching movies:', error);
    throw error;
  }
};

// Get movie image URL
export const getImageUrl = (posterPath) => {
  return posterPath ? `${IMAGE_BASE_URL}${posterPath}` : null;
};

File: Muzix_Auth_Service-main/Muzix_Auth_Service-main/src/main/java/com/muzix/auth/service/MovieService.java
package com.muzix.auth.service;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpMethod;
import org.springframework.http.ResponseEntity;

import java.util.HashMap;
import java.util.Map;

@Service
@RequiredArgsConstructor
public class MovieService {

    @Value("${tmdb.api.key:fdbc479e298584db44a1b74791d3c3f4}")
    private String tmdbApiKey;

    @Value("${tmdb.base.url:https://api.themoviedb.org/3}")
    private String tmdbBaseUrl;

    private final RestTemplate restTemplate = new RestTemplate();

    public Map<String, Object> getPopularMovies(String userEmail, String token) {
        String url = tmdbBaseUrl + "/movie/popular?api_key=" + tmdbApiKey;
        return fetchMoviesWithUserInfo(url, userEmail, token);
    }

    public Map<String, Object> getTrendingMovies(String userEmail, String token) {
        String url = tmdbBaseUrl + "/trending/movie/week?api_key=" + tmdbApiKey;
        return fetchMoviesWithUserInfo(url, userEmail, token);
    }

    public Map<String, Object> getTopRatedMovies(String userEmail, String token) {
        String url = tmdbBaseUrl + "/movie/top_rated?api_key=" + tmdbApiKey;
        return fetchMoviesWithUserInfo(url, userEmail, token);
    }

    public Map<String, Object> searchMovies(String query, String userEmail, String token) {
        String url = tmdbBaseUrl + "/search/movie?api_key=" + tmdbApiKey + "&query=" + query;
        return fetchMoviesWithUserInfo(url, userEmail, token);
    }

    private Map<String, Object> fetchMoviesWithUserInfo(String url, String userEmail, String token) {
        try {
            ResponseEntity<Map> response = restTemplate.getForEntity(url, Map.class);

            Map<String, Object> result = new HashMap<>();
            result.put("movies", response.getBody());
            result.put("userEmail", userEmail);
            result.put("timestamp", System.currentTimeMillis());

            return result;
        } catch (Exception e) {
            Map<String, Object> errorResult = new HashMap<>();
            errorResult.put("error", "Failed to fetch movies: " + e.getMessage());
            errorResult.put("userEmail", userEmail);
            return errorResult;
        }
    }
}